integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -203,9 +203,65 @@ public void shouldModifyProduceMessage(KafkaCluster cluster, Admin admin) throws
         }
     }
 
+    @Test
+    public void offsetManuallyManaged(KafkaCluster cluster, Admin admin) throws Exception {
+        admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+
+        String config = """
+                proxy:
+                  address: %s
+                clusters:
+                  demo:
+                    bootstrap_servers: %s
+                filters:
+                - type: ApiVersions
+                - type: BrokerAddress
+                """.formatted(PROXY_ADDRESS, cluster.getBootstrapServers());
+
+        try (var proxy = startProxy(config)) {
+            try (var producer = new KafkaProducer<String, String>(Map.of(
+                    ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, PROXY_ADDRESS,
+                    ProducerConfig.CLIENT_ID_CONFIG, "offsetManuallyManaged",
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000))) {
+                producer.send(new ProducerRecord<>(TOPIC_1, "my-key", PLAINTEXT)).get();
+                producer.send(new ProducerRecord<>(TOPIC_1, "my-key2", PLAINTEXT)).get();
+                producer.flush();
+            }
+
+            ConsumerRecords<String, String> records1;
+            long offsetCommitted;
+            long actualEndOffset;
+            try (var consumer = new KafkaConsumer<String, String>(Map.of(
+                    ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, PROXY_ADDRESS,
+                    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.GROUP_ID_CONFIG, "my-group-id",
+                    ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false",
+                    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+
+                Set<TopicPartition> partitions = new HashSet<>();
```

Comment:
I think this is redundant.

Reply:
What is redundant? Setting it to `earliest` or setting any value into `AUTO_OFFSET_RESET_CONFIG`?

If we don't set a value there, the consumer throws an exception because it needs to have a value there. [see here](https://medium.com/lydtech-consulting/kafka-consumer-auto-offset-reset-d3962bad2665)